src/move/econia/sources/market.move

```diff
@@ -628,6 +635,57 @@ module econia::market {
         map: Tablist<u64, OrderBook>
     }
 
+    struct MarketAccountInfo has copy, drop, store {
+        market_id: u64,
+        user: address,
+        custodian_id: u64
+    }
+
+    /// An event handle can be looked up via a view functions, then
+    /// passed into the REST API to query events emitted to the handle.
+    ///
+    /// Events are emitted to handle for the market, and to a handle for
+    /// affected user/custodian ID tuple (affected signer in the case of
+    /// a swap, which is is not affiliated with a market account).
+    struct MarketEventHandles has key {
+        /// match() (events deferred to calling function)
+        match_events_by_market:
+            Table<u64, EventHandle<MatchEvent>>,
+        match_events_by_market_account:
+            Table<MarketAccountInfo, EventHandle<MatchEvent>>,
+        /*
```

Comment:
Event handles for each account should live on the user's account, not the resource account. See Aptos's implementation of [`coin.move`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/coin.move) for what this should look like.

Reply:
See https://github.com/econia-labs/econia/pull/321#issuecomment-1622722806 for implementation per your recommendation.